### PR TITLE
Feature get comments by article

### PR DESCRIPTION
**Pull request by isramueka**
This pull request includes: 
- New endpoint, controller and model to fetch comments by article ID.
- New TDD to ensure the correct behaviour of the implementation.
- Refactor of the Middleware Error Handler.
- Refactor requested by previous Pull Request
- Added new error test case for Invalid article id
- Updated endpoint.json file with the new documentation

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -2,18 +2,38 @@ const express = require("express");
...
 
+app.get("/api/articles/:article_id/comments", getCommentsByArticle);
+
+// Refactored Middleware to handle custom errors as requested in (PR#4)
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    return res.status(err.status).send({ msg: err.msg });
+  }
+  next(err);
+});
```
Comment (nc-ali-abdallah) [inline app.js:25]: Awesome refactor.

File: app.js [hunk 2]
```diff
@@ -2,18 +2,38 @@ const express = require("express");
...
+    return res.status(400).send({ msg: "Invalid input" });
+  }
+  next(err);
+});
+
+//Refactor of err truthy, as it can handle any error
 app.use((err, req, res, next) => {
-  if (err.status === 404) {
+  if (err.status) {
     res.status(err.status).send({ msg: err.msg });
```
Comment (nc-ali-abdallah) [inline app.js:37]: our error handler above should catch this one.

File: models/articles.models.js [hunk 3]
```diff
@@ -14,4 +14,40 @@ const fetchArticleById = (article_id) => {
...
+    ORDER BY created_at DESC`,
+      [article_id]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: "No comments found for this article",
+        });
+      }
```
Comment (nc-ali-abdallah) [inline models/articles.models.js:48]: This isn't correct, if a comment has no article that is fine and we need to return it.
You'll need to come up with a separate way to check if the articleID exists .

**Review by nc-ali-abdallah — COMMENTED** [review]
Fantastic job of implementing feedback and the changes you've made look great.

Need some more tests for this task.
What if an invalid datatype is passed instead of the article_id?
Also I'd like to see a test for what if an article has so comments associated with it? should be a 200 and an empty array.

Could of things need correcting for this task but in general things are looking really good.
